pkg/ast/spl/spl.peg

```diff
@@ -671,6 +690,185 @@ GroupbyBlock <- BY fields:FieldNameList {
+    window, err := strconv.ParseUint(windowSize.(string), 10, 64)
+    if err != nil {
+        return nil, fmt.Errorf("Spl peg: Invalid window size: %v, must be a positive integer", windowSize.(string))
+    }
+    if window > 10000 {
+        return nil, fmt.Errorf("Spl peg: Invalid window size: %v, must be less than or equal to 10000", window)
+    }
+    ssOption := &structs.StreamStatsOptions{
+        Window: window,
+    }
+    ssOptionArg := &StreamStatsOptionArgs{
+        argOption: "window",
+        streamStatsOptions: ssOption,
+    }
+    
+    return ssOptionArg, nil
+}
+
+
+StreamStatsOptionResetBefore <- "reset_before" EQUAL L_PAREN boolExpr:BoolExpr R_PAREN {
+    ssOption := &structs.StreamStatsOptions{
+        ResetBefore: boolExpr.(*structs.BoolExpr),
+    }
+    ssOptionArg := &StreamStatsOptionArgs{
+        argOption: "reset_before",
+        streamStatsOptions: ssOption,
+    }
+    
+    return ssOptionArg, nil
+}
+
+StreamStatsOptionResetAfter <- "reset_after" EQUAL L_PAREN boolExpr:BoolExpr R_PAREN {
+    ssOption := &structs.StreamStatsOptions{
+        ResetAfter: boolExpr.(*structs.BoolExpr),
+    }
+    ssOptionArg := &StreamStatsOptionArgs{
+        argOption: "reset_after",
+        streamStatsOptions: ssOption,
+    }
+    
+    return ssOptionArg, nil
+}
+
+StreamStatsOptionTimeWindow <- "time_window" EQUAL spanLength:BinSpanLenOption {
+    spanLen := spanLength.(*structs.BinSpanLength)
+    if spanLen.TimeScale == utils.TMInvalid {
+        return nil, fmt.Errorf("Invalid Syntax, time_window option cannot be used without time scale")
+    }
+    ssOption := &structs.StreamStatsOptions{
+        TimeWindow: spanLen,
+    }
+    ssOptionArg := &StreamStatsOptionArgs{
+        argOption: "time_window",
+        streamStatsOptions: ssOption,
+    }
+    
+    return ssOptionArg, nil
+}
+
+StreamStatsOption <- ssOption:(StreamStatsOptionResetOnChange / StreamStatsOptionCurrent / StreamStatsOptionGlobal / StreamStatsOptionAllNum / StreamStatsOptionWindow / StreamStatsOptionResetBefore / StreamStatsOptionResetAfter / StreamStatsOptionTimeWindow) {
+    return ssOption, nil
+}
+
+StreamStatsOptionList <- first:StreamStatsOption rest:(SPACE StreamStatsOption)* {
+    restSlice := rest.([]any)
+    optionWasSpecified := make(map[string]bool)
+
+    numOptions := 1 + len(restSlice)
+
+    ssOption := initializeStreamStatsOptions()
+
+    for i := 0; i < numOptions; i++ {
+        var ssOptArg *StreamStatsOptionArgs
+        if i == 0 {
+            ssOptArg = first.(*StreamStatsOptionArgs)
+        } else {
+            separator := restSlice[i - 1].([]any)
+            ssOptArg = separator[1].(*StreamStatsOptionArgs)
+        }
+
+        argOption := ssOptArg.argOption
+        val, exist := optionWasSpecified[argOption]
+        if exist && val {
+            return nil, fmt.Errorf("Invalid Syntax, option: %v is already specified", argOption)
+        }
+        optionWasSpecified[argOption] = true
```

Comment:
You don't have to change this. I just want to point out that you have very similar code for option lists for other SPL commands, so it could be good at some point to make a general function to do this logic. That would reduce duplicate code and make it a little easier for you to add similar parts for handling other SPL commands since you'd just have to call a function